### PR TITLE
feat: Added support for django4.2

### PR DESCRIPTION
**Pull request by awais786**
Adding django4.2 support

https://github.com/edx/upgrades/issues/131

Executed both tools.

https://github.com/adamchainz/django-upgrade
https://github.com/browniebroke/django-codemod

### CODE REVIEW

**Comment by awais786** [comment]
https://docs.djangoproject.com/en/2.0/topics/http/middleware/#django.utils.deprecation.MiddlewareMixin

File: tox.ini [hunk 1]
```diff
@@ -14,9 +14,8 @@ addopts = --cov splash --cov-report term-missing --cov-report xml
 norecursedirs = .* docs requirements
 
 [testenv]
-deps = 
-    django32: Django>=3.2,<4.0
```
Comment (UsamaSadiq) [inline tox.ini:18]: We shouldn't remove `django32` yet.

File: .github/workflows/ci.yml [hunk 2]
```diff
@@ -16,7 +16,7 @@ jobs:
       matrix:
         os: [ubuntu-20.04]
         python-version: ['3.8']
-        toxenv: [quality, django32, django40]
+        toxenv: [quality, django32, django40, django42]
```
Comment (UsamaSadiq) [inline .github/workflows/ci.yml:19]: We should also update the coverage check to run against `django42` now to verify if that is also working perfectly or not.

**Review by UsamaSadiq — CHANGES_REQUESTED** [review]

File: CHANGELOG.rst [hunk 3]
```diff
@@ -16,7 +16,12 @@ Unreleased
 
 *
 
-[1.2.0] - 2022-01-26
```
Comment (UsamaSadiq) [inline CHANGELOG.rst:19]: This line should also be kept in the CHANGELOG since it is the header for previous release.

**Review by UsamaSadiq — APPROVED** [review]

**Pull request merged**